lib/exercism/user_finished_tracks.rb

```diff
@@ -24,9 +24,19 @@ def started_tracks
     ActiveRecord::Base.connection.execute(completed_count_sql).to_a
   end
 
+  def count_finished_problems(user_finished_exercises, track)
+    (user_finished_exercises & track.problems.map(&:slug)).size
+  end
+
+  def completed?(track_attributes)
```

Comment:
I think what you proposed is properly the best for now. 
I was thinking running `track = tracks.find { |t| t.id == db_row['track_id'] }` two times is not good. 
Sure, I have time to change this. Thank you for the code. 💯